model/AssessmentResultsService.php

```diff
@@ -0,0 +1,217 @@
+     */
+    public function getDeliveryData(\taoDelivery_models_classes_execution_DeliveryExecution $deliveryExecution)
+    {
+        $result = [
+            'start' => $deliveryExecution->getStartTime(),
+            'end' => $deliveryExecution->getFinishTime(),
+            'label' => $deliveryExecution->getLabel()
+        ];
+        return $result;
+    }
+
+
+    /**
+     * Get rubric to be printed
+     * Rubric is considered printed if it included to the section which has an item tagged by specified tag
+     * (@see )
+     *
+     * @param \taoDelivery_models_classes_execution_DeliveryExecution $deliveryExecution
+     */
+    public function getPrintableRubric(\taoDelivery_models_classes_execution_DeliveryExecution $deliveryExecution)
+    {
+        /** @var DeliveryService $deliveryService */
+        $deliveryService = $this->getServiceManager()->get('taoProctoring/delivery');
+        $session = $deliveryService->getTestSession($deliveryExecution);
+
+        $inputParameters = $deliveryService->getRuntimeInputParameters($deliveryExecution);
+        $testDefinition = \taoQtiTest_helpers_Utils::getTestDefinition($inputParameters['QtiTestCompilation']);
+        $sections = $testDefinition->getComponentsByClassName('assessmentSection');
+
+        $tag = $this->getOption(self::OPTION_PRINTABLE_RUBRIC_TAG);
+
+        $directoryIds = explode('|', $inputParameters['QtiTestCompilation']);
+        $compilationDirs = array(
+            'private' => \tao_models_classes_service_FileStorage::singleton()->getDirectoryById($directoryIds[0]),
+            'public' => \tao_models_classes_service_FileStorage::singleton()->getDirectoryById($directoryIds[1]),
+        );
+
+        $rubrics = [];
+
+        // -- variables used in the included rubric block templates.
+        // base path (base URI to be used for resource inclusion).
+        $basePathVarName = TAOQTITEST_BASE_PATH_NAME;
+        $$basePathVarName = $compilationDirs['public']->getPublicAccessUrl();
+
+        // state name (the variable to access to get the state of the assessmentTestSession).
+        $stateName = TAOQTITEST_RENDERING_STATE_NAME;
+        $$stateName = $session;
+
+        // views name (the variable to be accessed for the visibility of rubric blocks).
+        $viewsName = TAOQTITEST_VIEWS_NAME;
+        $$viewsName = array(View::CANDIDATE);
+
+        foreach ($sections as $section) {
+            $assessmentItemsRef = $section->getComponentsByClassName('assessmentItemRef');
+            foreach ($assessmentItemsRef as $item) {
+                foreach($item->getCategories() as $category) {
+                    if ($category === $tag) {
+                        foreach ($section->getRubricBlockRefs() as $rubric) {
+                            ob_start();
+                            include($compilationDirs['private']->getPath() . $rubric->getHref());
+                            $rubrics[] = ob_get_clean();
+                        }
```

Comment:
I would say this could go in a helper to make it cleaner as it is close to be the same code than in the QTI Test Driver.

However, about the technique, I don't see any other technical sugar enabling such a thing. If we look at usual template engines (Twig, Smarty, our own Template Renderer), we go for output buffering, there's no other clean way, AFAIK.

@jsconan Due to the relative urgence to have this feature working, I would ask you to merge this PR. @hutnikau, I would like you to think about putting this rendering code into an helper that proctoring and qti test can share in nice way.